app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,96 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Create a conversion host and enable it. This operation will run as an
+    # MiqTask.
+    #
+    # Both the 'resource_type' and 'resource_id' are mandatory arguments,
+    # and the 'resource_type' must be either 'Host' or 'VmOrTemplate'.
+    #
+    # You may optionally pass in 'param_v2v_vddk_package_url' or 'auth_user'
+    # arguments as well.
+    #
+    # POST /api/conversion_hosts {
+    #   "name": "some_name",
+    #   "resource_type": "Host",
+    #   "resource_id": "7"
+    #   "param_v2v_vddk_package_url": "some_url"
+    #   "auth_user": "some_user"
+    # }
+    #
+    def create_resource(type, id, data)
+      raise BadRequestError, "resource_id must be specified" unless data['resource_id']
+      raise BadRequestError, "resource_type must be specified" unless data['resource_type']
+
+      resource_type = data['resource_type']
```

Comment:
Should this also be deleted from the data array for the same reasons? https://github.com/ManageIQ/manageiq-api/pull/535/files#diff-2da9489e8f3f5aa77ab1c89559de7604R29

Reply:
No because you can't pass a resource object directly via the api. The `resource_type` is part of the table.

Reply:
It looks like the `enable_queue` method is expecting `:resource` and not `:resource_type`

Reply:
Oh, I think I see what you're saying. For `resource_id` and `resource_type` it doesn't hurt to leave them since they're still valid params to `ConversionHost.new`, whereas auth_user is not. Maybe it's inefficient, though? I guess I try not to mangle params if I can help it.

Reply:
`:resource_type` happens to be overridden in the core method, my concern is that the raw parameter is not constrained.

Reply:
This is now constrained by the whitelist.